### PR TITLE
feat: libsql experiment

### PR DESCRIPTION
**Pull request by markphelps**
- [libsql](https://libsql.org/about) support
- update to go 1.21

### CODE REVIEW

File: internal/storage/sql/db_test.go [hunk 1]
```diff
@@ -36,12 +36,21 @@ func TestOpen(t *testing.T) {
...
+				URL:             "file:/flipt.db",
 				MaxOpenConn:     5,
 				ConnMaxLifetime: 30 * time.Minute,
 			},
 			driver: fliptsql.SQLite,
 		},
+		{
+			name: "libsql url",
+			cfg: config.DatabaseConfig{
+				URL:             "libsql://file:/flipt.db",
```
Comment (markphelps) [inline internal/storage/sql/db_test.go:48]: FLIPT_DATABASE_URL='libsql://file:foo.db'

File: internal/storage/sql/db_test.go [hunk 2]
```diff
@@ -36,12 +36,39 @@ func TestOpen(t *testing.T) {
 		{
 			name: "sqlite url",
 			cfg: config.DatabaseConfig{
-				URL:             "file:flipt.db",
+				URL:             "file:/flipt.db",
```
Comment (markphelps) [inline internal/storage/sql/db_test.go:39]: is this backward compatible? ie does it also work with `file:flipt.db` ?

File: examples/database/libsql/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    image: ghcr.io/libsql/sqld:main
+    depends_on:
+      - sqld-primary
+    environment:
+      - SQLD_NODE=replica
+      - SQLD_PRIMARY_URL=http://sqld-primary:5001
+    volumes:
+      - "/tmp/replica.db:/var/lib/sqld/iku.db"
+  flipt-one:
+    image: flipt:dev-cb18b18
```
Comment (markphelps) [inline examples/database/libsql/docker-compose.yml:20]: im guessing this is from your Dagger build? maybe we can use our relative Dockerfile for now? at `../../../Dockerfile`? (project root)

Comment (yquansah) [reply examples/database/libsql/docker-compose.yml:20]: @markphelps Yeah meant to specify that point, apologies!

File: internal/storage/sql/db_test.go (hunk 2 above)
Comment (yquansah) [reply internal/storage/sql/db_test.go:39]: @markphelps Yes it seems to work with `file:flipt.db` as well.

Comment (markphelps) [reply internal/storage/sql/db_test.go:39]: can you add a test just to be sure?

Comment (yquansah) [reply internal/storage/sql/db_test.go:39]: @markphelps Yup!

File: Dockerfile.dev [hunk 4]
```diff
@@ -0,0 +1,26 @@
+FROM golang:1.21-alpine3.18
```
Comment (GeorgeMac) [inline Dockerfile.dev:1]: Is this one still needed?
I can't remember how that conversation ended.

**Review by GeorgeMac — APPROVED** [review]
This looks good to me. Only one question regarding the added Dockerfile.dev file.

File: Dockerfile.dev (hunk 4 above)
Comment (markphelps) [reply Dockerfile.dev:1]: i think it still is for the dev docker-compose to not build the UI in the Go binary

**Pull request merged**